pulse2percept/models/beyeler2019.py

```diff
@@ -904,6 +916,40 @@ def plot(self, use_dva=False, style='hull', annotate=True, autoscale=True,
             ann.set_yticks([])
         return ax
 
+class TorchAxonMapSpatial(torch.nn.Module):
+    def __init__(self, p2pmodel): # implant parameter currently not used
+        super().__init__()
+
+        # Check for model validity
+        if isinstance(p2pmodel, AxonMapModel) or not isinstance(p2pmodel, AxonMapSpatial):
+            raise ValueError("Must pass in a valid AxonMapModelSpatial")
+        if p2pmodel.engine != 'torch':
+            raise ValueError("Engine Selection Conflict : Constructing TorchAxonMapSpatial with engine", p2pmodel.engine)
```

Comment:
you should be able to construct a torchmodel from any axonmapspatial object, we can't guarantee that the torchmodel is being used internally within the p2pmodel
i.e. in the onnx conversion, we will most likely be passing in a p2pmodel with the default cython engine